lib/rspec/mocks/argument_matchers.rb

```diff
@@ -258,6 +268,36 @@ def formatted_expected_values
         end
       end
 
+      # @private
+      class ArrayExcludingMatcher
+        def initialize(unexpected)
+          @unexpected = unexpected
+        end
+
+        def ===(actual)
+          actual = actual.uniq
+          @unexpected.uniq.none? do |unexpected_element|
+            actual.any? do |actual_element|
+              RSpec::Support::FuzzyMatcher.values_match?(unexpected_element, actual_element)
+            end
+          end
```

Comment:
This implementation will be slow to pass, as it has to check the actual array repeadedly for each unexpected value, given that I'd expect the unexpected values to be shorter (although you could actually gate this check on the size and flip flop) I'd invert this to loop over actual once and check each value against all unexpected...

I'd also consider shortcutting with a plain intersection check, if that fails you can fail the example quickly.